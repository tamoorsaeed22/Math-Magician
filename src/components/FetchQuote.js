import React, { useState, useEffect } from 'react';

export default function DisplayQuote() {
  const [data, setData] = useState({});
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);

  useEffect(() => {
    async function fetchQuote() {
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=computers', {
          method: 'GET',
          headers: { 'X-Api-Key': 'a4jkLzxa11Fyl351KUe2lQ==OqEqrHKzRyti4wax' },
          contentType: 'application/json',
        });
        const result = await response.json();
        setData(result[0]);
        setLoading(false);
      } catch (error) {
        setError(true);
        setLoading(false);
        throw new Error(error);
      }
    }
    fetchQuote();
  }, []);

  if (loading) {
    return (
      <section className="quote_display">
        <h2>Quote</h2>
        <p>Loading quote...</p>
      </section>
    );
  }

  if (error || data === undefined) {
    return <p>Encountered a problem while retrieving the quote</p>;
  }

  return (
    <section className="quote_display">
      <h2>Quote</h2>
      <q>{data.quote}</q>
      <p className="author">{data.author}</p>
    </section>
  );
}
