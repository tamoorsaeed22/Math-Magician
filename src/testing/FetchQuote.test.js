import renderer from 'react-test-renderer';
import FetchQuote from '../components/FetchQuote';

it('renders quote correctly', () => {
  const tree = renderer.create(
    <FetchQuote />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
