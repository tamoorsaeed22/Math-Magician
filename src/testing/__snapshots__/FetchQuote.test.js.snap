// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote correctly 1`] = `
<section
  className="quote_fetch"
>
  <p
    style={
      Object {
        "fontSize": "1.3rem",
      }
    }
  >
    Loading quote...
  </p>
</section>
`;
